### PR TITLE
Update rollup to new major 3.6

### PR DESCRIPTION
**Pull request by susnux**
* Updated rollup to new major version 3.x (3.6.0) and also updated corresponding dependencies
* Adjusted tsconfig to run cypress using ES modules (ES2015 is the latest supported version by cypress 11)
* Fixes / Supersedes #94

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,18 +5,12 @@
...
 		"outDir": "./dist"
 	},
-	"ts-node": {
-		// these options are overrides used only by ts-node
-		// same as our --compilerOptions flag and our TS_NODE_COMPILER_OPTIONS environment variable
-		"compilerOptions": {
-			"module": "commonjs"
-		}
-	}
 }
```
Comment (skjnldsv) [inline tsconfig.json:16]: I think thi sis needed for the cypress testing
https://github.com/nextcloud/nextcloud-cypress/blob/master/cypress/dockerNode.ts

Comment (susnux) [reply tsconfig.json:16]: Locally it works perfect with Node 16, but it seems the github workflow does something different

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -30,8 +30,8 @@ jobs:
         uses: skjnldsv/read-package-engines-version-actions@v1.2
         id: versions
         with:
-          fallbackNode: '^12'
-          fallbackNpm: '^6'
+          fallbackNode: '^16'
+          fallbackNpm: '^7'
```
Comment (skjnldsv) [inline .github/workflows/cypress.yml:34]: Not really necessary :wink:

**Comment by susnux** [comment]
> Please do not upgrade cypress, this is broken #11

Had some hopes as at least 1 of 2 workflows succeeded

**Comment by skjnldsv** [comment]
> Had some hopes as at least 1 of 2 workflows succeeded

Me too. But this is also not related to this PR. Patchwork PRs should be avoided :)

File: .github/workflows/cypress.yml (hunk 2 above)
Comment (susnux) [reply .github/workflows/cypress.yml:34]: I think the workflows are updated from the organization repo anyways, so they should be updated there to fallback to the latest supported version.
I will undo that change :)

**Comment by susnux** [comment]
The cypress tests now succeed, the issue was related to module handling which changed in rollup `2 -> 3` and node support in cypress (cypress 12 dropped node 12 support and modernized the builds to support node 14 / 16 and 18).

I guess the Node errors are unrelated?

File: tsconfig.json (hunk 1 above)
Comment (susnux) [reply tsconfig.json:16]: works now! :rocket:

**Comment by susnux** [comment]
I had to remove the node 14 test as node 14 fails with `fsevents not accessible from chokidar`
But I think this is not that bad, as it is not supported anymore by nextcloud anyways.

File: package.json [hunk 3]
```diff
@@ -62,27 +62,26 @@
...
+    "@nextcloud/webpack-vue-config": "^5.4.0",
+    "@rollup/plugin-commonjs": "^23.0.3",
     "@rollup/plugin-node-resolve": "^15.0.1",
     "@rollup/plugin-typescript": "^10.0.1",
     "@types/cypress": "^1.1.3",
     "@types/dockerode": "^3.3.12",
     "@types/jest": "^29.2.2",
     "@vue/cli-service": "^5.0.8",
     "@vue/tsconfig": "^0.1.3",
-    "cypress": "^12.0.1",
```
Comment (skjnldsv) [inline package.json:74]: we lost cypress? :laughing:

Comment (skjnldsv) [inline package.json:74]: we lost cypress? :grin:

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (susnux) [reply package.json:74]: This is from when I updated cypress in this PR. Node / NPM starting with NPM 7 installs all peer dependencies automatically when installing a package (e.g. for developing / testing).
So all dependencies are only required once in either dependencies, devDependencies or peerDependencies.

But if you like keeping the dependency twice to make clear it is also a devDependency I revert this :smile:

Comment (susnux) [reply package.json]: Edit: I reverted this as this is unrelated to this PR so do not mix up topics here :sweat_smile:

Comment (skjnldsv) [reply package.json]: > But if you like keeping the dependency twice to make clear it is also a devDependency I revert this

Yes, we do this. peerDependencies AND devDependencies is required. They serve different purpose

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**